library/std/src/sys/uefi/common.rs

```diff
@@ -0,0 +1,493 @@
+use r_efi::efi::{EventNotify, Guid, Tpl};
+
+use crate::alloc::{AllocError, Allocator, Global, Layout};
+use crate::ffi::{OsStr, OsString};
+use crate::io::{self, const_io_error};
+use crate::mem::{self, MaybeUninit};
+use crate::os::uefi;
+use crate::os::uefi::ffi::{OsStrExt, OsStringExt};
+use crate::ptr::NonNull;
+
+pub(crate) const BOOT_SERVICES_ERROR: io::Error =
+    const_io_error!(io::ErrorKind::Other, "failed to acquire boot services",);
+pub(crate) const RUNTIME_SERVICES_ERROR: io::Error =
+    const_io_error!(io::ErrorKind::Other, "failed to acquire runtime services",);
+
+/// Get the Protocol for current system handle.
+/// Note: Some protocols need to be manually freed. It is the callers responsibility to do so.
+pub(crate) fn get_current_handle_protocol<T>(protocol_guid: Guid) -> Option<NonNull<T>> {
+    let system_handle = uefi::env::image_handle();
+    open_protocol(system_handle, protocol_guid).ok()
+}
+
+#[repr(transparent)]
+pub(crate) struct Event {
+    inner: NonNull<crate::ffi::c_void>,
```

Comment:
Yes, that seems to be case after asking in the edk2 mailing list: https://edk2.groups.io/g/devel/topic/94092613#94674

Reply:
IIUC, she is referring to the pointer the event is returned through, but not the event pointer itself. I don't think that we can safely assume that the event pointer is non-null. For instance, I can image implementations returning an 0-based index instead of a pointer, which is allowed by the specification but would break `NonNull`.

Reply:
Check out the next reply in the list: https://edk2.groups.io/g/devel/message/94678
But yes, it might be possible to have a Null pointer for Event since the spec does not forbid it.

Reply:
In that case, this should be a raw pointer, not `NonNull`.